homeassistant/components/tasmota/fan.py

```diff
@@ -0,0 +1,93 @@
+"""Support for Tasmota fans."""
+
+from hatasmota import const as tasmota_const
+
+from homeassistant.components import fan
+from homeassistant.components.fan import FanEntity
+from homeassistant.core import callback
+from homeassistant.helpers.dispatcher import async_dispatcher_connect
+
+from .const import DATA_REMOVE_DISCOVER_COMPONENT, DOMAIN as TASMOTA_DOMAIN
+from .discovery import TASMOTA_DISCOVERY_ENTITY_NEW
+from .mixins import TasmotaAvailability, TasmotaDiscoveryUpdate
+
+HA_TO_TASMOTA_SPEED_MAP = {
+    fan.SPEED_OFF: tasmota_const.FAN_SPEED_OFF,
+    fan.SPEED_LOW: tasmota_const.FAN_SPEED_LOW,
+    fan.SPEED_MEDIUM: tasmota_const.FAN_SPEED_MEDIUM,
+    fan.SPEED_HIGH: tasmota_const.FAN_SPEED_HIGH,
+}
+
+TASMOTA_TO_HA_SPEED_MAP = {
```

Comment:
Optionally, could do something like:

TASMOTA_TO_HA_SPEED_MAP = {v: k for k, v in HA_TO_TASMOTA_SPEED_MAP.items()}